packages/bibliography/src/main.rs

```diff
@@ -0,0 +1,358 @@
+        // doesn't, in this version at least). We encase it in [] for IEE and () for others
+        let json_citation = if style_arg == "IEEE" {
+            let mut vec = vec![text!("[")];
+            vec.append(&mut display(&database_citation.display));
+            vec.push(text!("]"));
+            Value::Array(vec)
+        } else {
+            let mut vec = vec![text!("(")];
+            vec.append(&mut display(&database_citation.display));
+            vec.push(text!(")"));
+            Value::Array(vec)
+        };
+
+        // Then, we are adding that citation as the label (that the [cite], now [cite-internal],
+        // will turn into). Note, since we are doing string-matching later on (not value-based
+        // matching), we want to make sure not to parse citation_str and re-stringify it since
+        // object kv-pairs may change order. The string must match exactly to be picked up again
+        output.push(add_citation_label!(
+            citation_str,
+            format!("{json_citation}")
+        ));
+    }
+
+    let visibility = input["arguments"]["visibility"].as_str().unwrap();
+
+    // If the bibliography should be shown, show it!
+    if visibility == "shown" {
+        for entry in database.bibliography(bibliography_style.as_ref(), None) {
+            if let Some(prefix) = &entry.prefix {
+                // If we have a prefix (which is the number in [] for IEEE), encase it in brackets
+                output.push(text!("["));
+                output.append(&mut display(prefix));
+                output.push(text!("] "));
+            }
+            output.append(&mut display(&entry.display));
+            output.push(module!("newline", ""));
+        }
+    }
+
+    println!("{}", Value::Array(output));
+}
+
+/// Gets the styles for the given key, as a pair of the bibliography style and citation style. If
+/// the key has no style mappings, None is returned. For the keys IEEE, APA, MLA and CMoS, Some is
+/// always returned, and for any other key, None is returned.
+fn get_styles(key: &str) -> Option<(Box<dyn BibliographyStyle>, Box<dyn CitationStyle>)> {
+    let bibliography_style: Box<dyn BibliographyStyle> = match key {
+        "IEEE" => Box::new(Ieee::new()),
+        "APA" => Box::new(Apa::new()),
+        "MLA" => Box::new(Mla::new()),
+        "CMoS" => Box::new(ChicagoAuthorDate::new()),
+        _ => return None,
+    };
+
+    let citation_style: Box<dyn CitationStyle> = match key {
+        "IEEE" => Box::new(Numerical::new()),
+        "APA" => Box::new(ChicagoAuthorDate::new()),
+        "MLA" => Box::new(ChicagoAuthorDate::new()),
+        "CMoS" => Box::new(ChicagoAuthorDate::new()),
+        _ => return None,
+    };
+
+    Some((bibliography_style, citation_style))
+}
+
+/// This function reads the bibliography by first:
+/// * Checking the 'file' argument, and if it is non-empty, reads that file
+/// * Otherwise, checking the body of the module
```

Comment:
Suggestion: perhaps we could append the content of the [bibliography] to the file's content if both are defined instead of just ignoring one of them.

Reply:
Hmm, I don't know if this would be good and useful or just confusing to users. But that is a nice suggestion, maybe we should think about that some more.

Reply:
Added a warning if we supply both a body and file. We can change this behaviour later on if needed.